### PR TITLE
Update JS dependencies for api augment and e2e tests

### PR DESCRIPTION
**Pull request by wilwade**
# Goal
The goal of this PR is to get Api Augment to Polkadotjs v11+

# Discussion
- Update eslint to flat config files and ready for v9
- Polkadotjs/api v11
- Node to v20.12.2
- Other small package updates

### CODE REVIEW

File: e2e/eslint.config.mjs
Comment (wilwade) [inline e2e/eslint.config.mjs:1]: Switching to the new flat config eslint style

File: e2e/load-tests/signatureRegistry.test.ts
Comment (wilwade) [inline e2e/load-tests/signatureRegistry.test.ts:1]: All automatic formatting and linting

File: e2e/scaffolding/extrinsicHelpers.ts [hunk 1]
```diff
@@ -23,46 +24,46 @@ import { u8aWrapBytes } from '@polkadot/util';
 import type { Call } from '@polkadot/types/interfaces/runtime';
 import { hasRelayChain } from './env';
```
Comment (wilwade) [inline e2e/scaffolding/extrinsicHelpers.ts:26]: All automatic formatting and linting below

File: e2e/scaffolding/extrinsicHelpers.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import '@frequency-chain/api-augment';
```
Comment (wilwade) [inline e2e/scaffolding/extrinsicHelpers.ts:1]: Not sure how this was working without this before, but it has to be called before polkadot/api

File: e2e/scaffolding/helpers.ts
Comment (wilwade) [inline e2e/scaffolding/helpers.ts:1]: All automatic formatting and linting

File: js/api-augment/eslint.config.mjs
Comment (wilwade) [inline js/api-augment/eslint.config.mjs:1]: Tried to keep it as same as possible

File: js/api-augment/package.json [hunk 3]
```diff
@@ -27,29 +27,26 @@
     "generate:meta": "node --import=tsx/esm node_modules/.bin/polkadot-types-from-chain --package @frequency-chain/api-augment --endpoint ./metadata.json --output ./interfaces",
     "test": "mocha --import=tsx  ./test/**/*.test.ts",
     "test:only": "mocha --import=tsx",
-    "format": "tsc --noEmit --pretty && eslint --fix \"**/*.ts\"",
-    "lint": "tsc --noEmit --pretty && eslint \"**/*.ts\""
+    "format": "tsc --noEmit --pretty && prettier --write . && eslint --fix",
+    "lint": "tsc --noEmit --pretty && prettier --check . && eslint"
```
Comment (wilwade) [inline js/api-augment/package.json:31]: eslint doesn't do the prettier side anymore

File: js/api-augment/scripts/package.cjs
Comment (wilwade) [inline js/api-augment/scripts/package.cjs:1]: All automatic formatting and linting

File: js/api-augment/scripts/pre-gen.cjs
Comment (wilwade) [inline js/api-augment/scripts/pre-gen.cjs:1]: All automatic formatting and linting

File: js/api-augment/test/index.test.ts
Comment (wilwade) [inline js/api-augment/test/index.test.ts:1]: All automatic formatting and linting

File: js/api-augment/tsconfig.cjs.json
Comment (wilwade) [inline js/api-augment/tsconfig.cjs.json:1]: All automatic formatting and linting

File: js/api-augment/tsconfig.eslint.json
Comment (wilwade) [inline js/api-augment/tsconfig.eslint.json:1]: All automatic formatting and linting

File: js/api-augment/tsconfig.esm.json
Comment (wilwade) [inline js/api-augment/tsconfig.esm.json:1]: All automatic formatting and linting

File: js/api-augment/tsconfig.json
Comment (wilwade) [inline js/api-augment/tsconfig.json:1]: All automatic formatting and linting

File: js/api-augment/tsconfig.types.json
Comment (wilwade) [inline js/api-augment/tsconfig.types.json:1]: All automatic formatting and linting

File: e2e/package.json [hunk 4]
```diff
@@ -19,27 +19,26 @@
...
     "@polkadot/util": "12.6.2",
-    "helia": "^4.1.0",
+    "helia": "^4.2.1",
     "multiformats": "^13.1.0",
     "rxjs": "^7.8.1",
-    "workerpool": "^9.1.0"
+    "workerpool": "^9.1.1"
   },
   "devDependencies": {
-    "@polkadot/typegen": "10.12.4",
```
Comment (wilwade) [inline e2e/package.json:32]: Not sure why this had this dependency. It isn't used.

File: js/api-augment/scripts/pre-gen.cjs [hunk 5]
```diff
@@ -6,18 +6,18 @@ const fs = require("fs");
 // eslint-disable-next-line
 const path = require("path");
 
-const interfacePath = path.join(__dirname, '../interfaces');
+const interfacePath = path.join(__dirname, "../interfaces");
```
Comment (mattheworris) [inline js/api-augment/scripts/pre-gen.cjs:9]: nit: wait, is it `'` or `"`? I just had social-app-template change a bunch to `'` 🤷‍♂️

**Review by mattheworris — APPROVED** [review]
- Read through all the changes, looks good.
lgtm

**Review by shannonwells — APPROVED** [review]
* Read through the changes
* Checked out branch & ran `make e2e-tests`, which passed.
Good to go pending CI passing :)
:shipit:

File: js/api-augment/scripts/pre-gen.cjs (hunk 5 above)
Comment (wilwade) [reply js/api-augment/scripts/pre-gen.cjs]: Good catch. I had missed making sure that the prettier config was the same for both e2e and api-augment. Our default is to do single quotes. Updated.

**Review by JoeCap08055 — APPROVED** [review]

**Pull request merged**